katello/katello/hostname-change.rb

```diff
@@ -369,18 +381,30 @@ def run
       installer << " --disable-system-checks" if disable_system_check_option?
 
       STDOUT.puts installer
-      installer_output = self.run_cmd("#{installer}")
-      installer_success = $?.success?
-      STDOUT.puts installer_output
-
-      STDOUT.puts "Restarting puppet services"
-      self.run_cmd("/sbin/service puppet restart")
-      self.run_cmd("katello-service restart --only puppetserver")
+      default_installer << ' --disable-system-checks' if disable_system_check_option?
+      failure_message = """
+Something went wrong with the #{@options[:scenario].capitalize} installer.
+Please check the above output and the corresponding logs. Once the issue is resolved
+you may complete the hostname change by running #{default_installer}
+      """
+
+      run_cmd(installer, [0], failure_message) do |result, success|
+        if temp_last_scenario_yaml && temp_last_scenario_yaml.path
+          unless success
+            STDOUT.puts 'restoring last_scenario.yaml'
+            run_cmd("cp #{temp_last_scenario_yaml.path} #{last_scenario_yaml}")
+          end
+          STDOUT.puts 'cleaning up temporary files'
+          temp_last_scenario_yaml.unlink
+        end
 
-      if installer_success
-        self.successful_hostname_change_message
-      else
-        self.fail_with_message("Something went wrong with the #{@options[:scenario].capitalize} installer! Please check the above output and the corresponding logs")
+        if success
+          STDOUT.puts result
+          STDOUT.puts 'Restarting puppet services'
+          run_cmd('systemctl reload-or-try-restart puppet')
```

Comment:
This should be try-reload-or-restart:
```
  try-reload-or-restart NAME...   If active, reload one or more units,
                                  if supported, otherwise restart
```

If you want to restart both puppet and puppetserver then you can even do it at once with systemctl (and in parallel):
```systemctl try-reload-or-restart puppet puppetserver```

Reply:
Should it? I do not see that option on my EL7 nor my Centos box. It also looks like they are equivalent methods in any case:

https://github.com/systemd/systemd/blob/3e52ec43011f0b7f72c5bf43ce60fa855d1a0ee2/src/systemctl/systemctl.c#L2784-L2785

Reply:
That does appear to be a difference between Fedora and CentOS. If this works on EL7 then let's go with this.